editors/sc-ide/widgets/settings/sclang_page.cpp

```diff
@@ -64,6 +64,8 @@ SclangPage::SclangPage(QWidget* parent): QWidget(parent), ui(new Ui::SclangConfi
 
     connect(ui->sclang_post_inline_warnings, SIGNAL(stateChanged(int)), this, SLOT(markSclangConfigDirty()));
     connect(ui->sclang_exclude_default_paths, SIGNAL(stateChanged(int)), this, SLOT(markSclangConfigDirty()));
+
+    connect(ui->sclang_port, SIGNAL(valueChanged(int)), this, SLOT(markSclangConfigDirty()));
```

Comment:
Sorry, I didn't fully follow the logic - would this trigger writing to the config file also, even though no changes were made?

Reply:
Yeah, but what I'm saying is that it would be best not to overwrite the sclang.conf if we don't actually change _that_.

Reply:
I was specifically talking about sclang.conf - that only has include/exclude paths etc. IIUC.
The port number is written in the ide conf file (right?) and yes, that one should be overwritten. 
But I didn't check the logic closely whether both files get overwritten at the same time or only when relevant changes happen?

Reply:
now I see what you are talking about - I indeed rewrote the sclang_conf file, but I don't think that this is a big problem in the bigger scope of the IDE ;)

I adjusted the PR now such that changing the port number will trigger the "restart language?" dialog but not write the sclang_conf file.
I did some code maintenance which makes the things a bit clearer.

Once we have the new connect syntax when dropping Qt6 we can also use lambda functions which should reduce the necessary methods.

While I were at it - I prefixed the member variables with `m`.

Reply:
I tested it but please verify if you also think that the behavior is as expected :)